config.go

```diff
@@ -8,86 +8,66 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-// ConfigFileName is an optional file checked for in the function root.
-const ConfigFileName = ".faas.yaml"
+// ConfigFileName is the name of the config's serialized form.
+const ConfigFileName = ".faas.config"
 
-// Config object which provides another mechanism for overriding client static
-// defaults.  Applied prior to the WithX options, such that the options take
-// precedence if they are provided.
+// Config represents the serialized state of a Function's metadata.
+// See the Function struct for attribute documentation.
 type Config struct {
-	// Name specifies the name to be used for this function. As a config option,
-	// this value, if provided, takes precidence over the path-derived name but
-	// not over the Option WithName, if provided.
-	Name string `yaml:"name"`
-
-	// Runtime of the implementation.
-	Runtime string `yaml:"runtime"`
-
-	// OCI image tag for the function
-	// typically of the form "registry/namespace/repository:tag"
-	Tag string `yaml:"tag"`
-
-	// Add new values to the applyConfig function as necessary.
+	Name      string `yaml:"name"`
+	Namespace string `yaml:"namespace"`
+	Runtime   string `yaml:"runtime"`
+	Image     string `yaml:"image"`
+	// Add new values to the toConfig/fromConfig functions.
 }
 
-// newConfig creates a config object from a function, effectively exporting mutable
-// fields for the config file while preserving the immutability of the client
-// post-instantiation.
-func newConfig(f *Function) Config {
-	return Config{
-		Name:    f.Name,
-		Runtime: f.Runtime,
-		Tag:     f.Tag,
+// newConfig returns a Config populated from data serialized to disk if it is
+// available.  Errors are returned if the path is not valid, if there are
+// errors accessing an extant config file, or the contents of the file do not
+// unmarshall.  A missing file at a valid path does not error but returns the
+// empty value of Config.
+func newConfig(root string) (c Config, err error) {
+	filename := filepath.Join(root, ConfigFileName)
+	if _, err = os.Stat(filename); os.IsNotExist(err) {
+		err = nil // do not consider a missing config file an error
+		return    // return the zero value of the config
 	}
```

Comment:
What if `err` is some error other than not existing?

Reply:
Correct, I must have confused the check two lines down as pertaining to that error.